### PR TITLE
Add published flag to model tables

### LINKED ISSUE #383: Add published flag column to all user-writable tables
VERY IMPORTANT: add a spec that tests if there are no tables which could be populated by user, either via the submission system, or via API, and that do not have the new published flag.

Default all current data to published, not private.

*OR* add the flag field to *ALL* tables (which makes the spec mentioned above not required). This might be more practical option.

### PR DESCRIPTION
**Pull request by nowakowski**
Fixes #383

### CODE REVIEW

**Comment by nowakowski** [comment]
Also fixes some old migrations to enable the test DB to be built from scratch.

**Comment by Nuanda** [comment]
@kammerer Tomasz, perhaps you should do that review since you are going to use it.

File: db/migrate/20160311121620_add_published_flag_to_db_columns.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    :qtl_jobs,
+    :restriction_enzymes,
+    :submission_uploads,
+    :submissions,
+    :taxonomy_terms,
+    :trait_descriptors,
+    :trait_grades,
+    :trait_scores,
+    :users
+  ]
```
Comment (kammerer) [inline db/migrate/20160311121620_add_published_flag_to_db_columns.rb]: I am not convinced we need `:users`, `:api_keys` or `:countries` here.

**Comment by kammerer** [comment]
:+1: Except for one minor comment.

**Comment by Nuanda** [comment]
Correct, these three tables could be excluded from the list. :+1:

**Comment by nowakowski** [comment]
Fixed. @kammerer - can I merge this now?

**Comment by kammerer** [comment]
Sure.
12 mar 2016 16:26 "Piotr Nowakowski" <notifications@github.com> napisał(a):

**Comment by Nuanda** [comment]
@nowakowski The schema file is wrong. You changed the migration but you did not rebuild the schema. Please do.

**Pull request merged**